test/ProgressPlugin.test.js

```diff
@@ -20,21 +21,61 @@ const createMultiCompiler = () => {
 };
 
 describe("ProgressPlugin", function() {
-	it("should not contain NaN as a percentage when it is applied to MultiCompiler", function(done) {
+	let _env;
+	let stderr;
+
+	beforeEach(() => {
+		_env = process.env;
+		stderr = Stdio.capture(process.stderr);
+	});
+	afterEach(() => {
+		process.env = _env;
+		stderr && stderr.restore();
+	});
+
+	it("should not contain NaN as a percentage when it is applied to MultiCompiler", () => {
 		const compiler = createMultiCompiler();
 
-		let percentage = 0;
-		new webpack.ProgressPlugin((p, msg, ...args) => {
-			percentage += p;
-		}).apply(compiler);
-
-		compiler.run(err => {
-			if (err) {
-				throw err;
-			} else {
-				expect(percentage).not.toBe(NaN);
-				done();
-			}
+		new webpack.ProgressPlugin().apply(compiler);
+
+		return RunCompilerAsync(compiler).then(() => {
+			expect(stderr.toString()).not.toContain("NaN");
 		});
 	});
+
+	it("should not print lines longer than stderr.columns or 40", () => {
+		const compiler = webpack({
+			context: path.join(__dirname, "fixtures"),
+			entry: "./a.js"
+		});
+
+		compiler.outputFileSystem = new MemoryFs();
+
+		new webpack.ProgressPlugin().apply(compiler);
+
+		process.stderr.columns = 10;
+
+		return RunCompilerAsync(compiler)
+			.then(() => {
+				const logs = stderr
+					.toString()
+					.split(/+/)
+					.filter(v => !(v === " "));
+
+				expect(logs.length).toBeGreaterThan(20);
+				logs.map(v => expect(v.length).toBeLessThanOrEqual(10));
```

Comment:
weird use of `Array.map` here. Either use `forEach` or better use a `for of` loop.